### ISSUE TITLE
Retraining an existing model with new data.

### ISSUE THREAD

**Issue by ragrawal26** [sdv-dev/SDV#1726]
### Environment details

If you are already running SDV, please indicate the following details about the environment in
which you are running it:

* SDV version: 1.2.0
* Python version: 3.9
* Operating System: Windows 

### Problem description

We have been trying to use HMA Synthesizer to train and generate multi-table synthetic data as a POC for our project.
We were able to generate the data successfully with good metrics however we are stuck now as the ask is to train the same model (trained above and saved) with new/delta of real data.

We are unable to see any option to train the same model with new data , without needing to train it again on older one.

Could you please help here?

Also please let us know if this is something which could be a part of enterprise version.

Thanks!

**Comment by npatki** [comment]
Hi @ragrawal26, nice to meet you.

Just to make sure I understand correctly, you have already trained an HMASynthesizer on some data and you would like to update the already-learned parameters based on new data (delta). Is that correct?

Currently, our synthesizers do not support updating (aka batch training), though there is an ongoing feature request for it in #805. You can follow it if you'd like to track any updates.

In the meantime, could you describe more about your use case? I'm curious about this requirement in the POC. How often do you expect to be updating your model and for which reasons? Our users find that this requirement is no longer needed if the synthesizer is fast. For example, the [HSASynthesizer](https://docs.sdv.dev/sdv/multi-table-data/modeling/synthesizers/hsasynthesizer) is able to model data in a matter of a few minutes (if not seconds) -- so it is not a big deal to train an entirely new HSA synthesizer with old + new data.

**Comment by ragrawal26** [comment]
Hello Neha,

Thanks for your response!

Yes, your understanding is correct, we are looking to train the model on new/delta data on already trained data to update the parameters.

We expect model to be trained on new data based on releases which may range between 2 months to 3months interval.

Since we can have the access to old data only once/twice and cannot keep the data store for more than a week(due to complaince), we have a limitation to train the model with old data + new data.

Hence we are looking for something like retraining the model on delta rather than a complete set.

Please let us know if there is any other way to achieve this.

Thanks!

**Comment by npatki** [comment]
Hi @ragrawal26 thanks for the explanation. Makes sense to me. Unfortunately, there is no easy workaround for this -- the team would have to support the batch training feature, which requires some core algorithmic updates.

I'm also thinking more about your particular usage. How much new data is available to you during every release? Our [research shows](https://datacebo.com/blog/sdv-training-subsample/) that as few as 2.5K rows are enough for getting decent quality data. This would indicate that training a new synthesizer for each release may not be a bad approach -- or at least, continuing to train on every delta may not provide as much value. Of course, this depends on how your data is expected to change.

Another factor is whether you expect the schema of your application to change over time? In this case, you would have to create a new synthesizer regardless, as the metadata would evolved.

**Comment by npatki** [comment]
Hi @ragrawal26, I'm closing this thread since we answered the original question and that there is an ongoing feature request for it in #805.

Please feel free to reach out here if you'd like to continue discussing your project, or any thoughts on the questions I've asked above. I can always re-open the issue for continued conversation. Thanks.
